actions/src/email_actions.py

```diff
@@ -1,13 +1,34 @@
-from typing import Callable, Dict, List
+from dataclasses import dataclass
+from typing import Callable, Dict, List, Union
 from email_api.email_api import EmailApi
 from openstack_api.openstack_floating_ip import OpenstackFloatingIP
+from openstack_api.openstack_image import OpenstackImage
 from openstack_api.openstack_query import OpenstackQuery
 from openstack_api.openstack_server import OpenstackServer
 
 from st2common.runners.base_action import Action
 
 
 class EmailActions(Action):
+    @dataclass
+    class EmailActionParams:
+        """
+        Structure containing the information needed to _email_users for a particular OpenstackResource
+        :param: required_email_property: The name of the property that must be obtained to get the email of the

```

Comment:
So the dataclass doesn't check if the property is not in the properties to select. This snippet of doc-string should be moved to where that check is actually done.

Ditto on the errors too, since this is a plain ol' dumb data class with no additional checking